xdp-document.c

```diff
@@ -181,6 +181,40 @@ xdp_document_get_permissions (XdpDocument *doc,
   return flags;
 }
 
+static void
+permissions_saved (GObject *source_object,
+                   GAsyncResult *result,
+                   gpointer data)
+{
+  GomResource *resource = GOM_RESOURCE (source_object);
+  GError *error = NULL;
+
+  if (!gom_resource_save_finish (resource, result, &error))
+    {
+      g_warning ("Failed to save permissions: %s", error->message);
+      g_error_free (error);
+    }
+}
+
+void
+xdp_document_grant_permissions (XdpDocument *doc,
+                                const char *app_id,
+                                XdpPermissionFlags perms)
+{
+  XdpPermissions *permissions = NULL;
+  XdpPermissionFlags old_perms;
+  GomRepository *repo;
+
+  old_perms = xdp_document_get_permissions (doc, app_id);
+  if ((perms & old_perms) == perms)
+    return;
```

Comment:
Actually I'm not quite sure this is right. I see it does not do this atm, but at some point I was thinking that permission granting should return the permission unique id so that one can revoke this grant, but if the app had a grant from elsewhere it would keep that. This means we have to always add each permission object independent of what was there before. Does that make sense?